### ISSUE TITLE
usage of external temperature sensor

### ISSUE THREAD

**Issue by turbolooser** [rospogrigio/daikin_residential#169]
First a big thank you for this Integration. Works like a charm for me.

My problem is just that the integrated temperature sensor in my Perfera devices is not reliable enough fo a good temperature management in my rooms. So I would like to use an external temperature sensor for the proper temperature but IO cannot get this to work somehow. All Thermostat integrations that I have tried by now are targetting heater/cooler switches which do not work with this Integration so I am wondering if this external sensor capability could be implemented or if anyone else does have a working idea/suggestion to this problem.

Thank you
Thomas

**Comment by jmtatsch** [comment]
Glad I am not the only one suffering from the Daikin Perfera.
Unfortunately I think it's not the temperature sensor but the control loop they have.

Here an example which I am experiencing right now:
Perfera setpoint is 23,5 °
Perfera thinks its 22°
the external thermometer at other side of room says 21,3°
Now the device blows cold air out instead of heating further 🤬

**Comment by turbolooser** [comment]
> Glad I am not the only one suffering from the Daikin Perfera. Unfortunately I think it's not the temperature sensor but the control loop they have.
> 
> Here an example which I am experiencing right now: Perfera setpoint is 23,5 ° Perfera thinks its 22° the external thermometer at other side of room says 21,3° Now the device blows cold air out instead of heating further 🤬

If the target is 23.5 and the perfera measures 22 then it should rather heat than blow cold air. I do confirm that that the behaviour is kind of weird also for me. I would love to use any other thermostat from hassio(or custom) which could manage this better. Any ideas are highly welcome

**Comment by rospogrigio** [comment]
Are you using the Auto mode? The guy who installed my appliance suggested me to either use the Cool or the Heat mode so you tell the unit what it shall do, and not let it decide.

**Comment by jmtatsch** [comment]
Yes, this is using auto mode. Also using heating only it's difficult to get a comfortable temperature. It's always either too hot or cold because it never heats to the set temperature but stays within like 2 degrees of it 😡

**Comment by rgerhards** [comment]
For a single unit, auto mode is fine! For multiple indoor units, it can be problematic, as all units must share the mode.

**Comment by turbolooser** [comment]
You are right @rgerhards . I have 3 indoor units and its a mess on all these conflicting modes. I am fine to run all units in heat or cool mode rather than auto. As winter and heating time approaches I am trying to use better thermostat now. Lets see what it can do for heating and if we do have a "cool" mode option available in 2024 when its getting hot again :-D

**Comment by Threesa** [comment]
Daikin should just add a temperature range option instead of fixed set temperature.
For example 21-23 °C. 

For one it allows for a comfort zone, thus not wasting energy too much to keep a specific value.
And less chance in auto mode for conflicting devices due to heating vs cooling.

**Comment by turbolooser** [comment]
Better Thermostat seems to have its own issues. Anyone having a good solution to this control issue?
Which Thermostat are you using to run your Daikin on schedule and with appropriate Temps set?

**Comment by OcinO88** [comment]
I just use a custom automation, where I use a separate temp sensor as trigger (if value gets below or above xx), for conditions I want my phone to be home and based upon the outside temp (daikin sensor) I set it to cool or heat. In case of cooling I use the target_temp_high and target_temp_low setting to define the max/min temp, but also use my temp sensor's value -3 degrees (celcius) to set the target temp for the airco unit (so it isn't extremely cold inside and when going outside, I don't have such a big temp difference). In case of heating I just use a static value.

**Comment by BeeGee1986** [comment]
I have 6 indoor Daikin units, all connected in the Onecta app. I use external Zigbee temp sensors and use the "Simple Thermostat" module available in HACS. In this module, I've define my Zigbee sensors as the only "source of truth". I've even disabled the temperature entities coming from this Daikin integration.

I've done some further testing and can't fully get my head around this issue, but so far my preliminary conclusion is the following: No matter what you do, Daikin Onecta will overrule what you define in HA if it thinks the desired temperature has been reached.

I accidently found this out with the following scenario:

- Target temp set to 22° via HA
- External Zigbee sensor reports 19° 
- Daikin thinks it is 25°

(The massive difference between my Zigbee temp and Daikin temp was because my sensor was far away, behind a corner and was lying on the floor accidently)

Result: the unit stopped heating. It didn't shut down, it just stops blowing air as like in "pause" mode.

I've done some further testing with this and it seems Daikin takes a  marging of 3° difference between what it reads as temp vs what is set as target. If the temperate is set to 22° and it's own sensor reads 25°, it will stop blowing air.

I'm not technical enough to dig into the code, but from what I can see, what this integration does is that it sets a target temp, however, it doesn't tell Daikin to listen to another sensor or disable its own sensor (probably Daikin doesn't even allow you to do this?)

Imho, as long as there is no way to override Daikin's temp sensor or disable it, there is no real solution... Unless someone has a creative idea on how we can "trick" the internal temp sensor of the Daikin unit?

**Comment by jwillemsen** [comment]
This integration only sets the target temperature, all control logic is within the daikin unit

**Comment by OcinO88** [comment]
That's why I created my own automation. If I want it to be 20 degrees C, I'll set it to heat and to 23 or 24 degrees. Then I have another automation that will stop the airco if the external temp sensor reaches 20 (actually, I use the average temp, from 2 temp sensors in my room). The first automation will start it when the temp gets below 18, so this way the unit will only be on until the temp is reached and I bypass Daikin logic that way, and also their temp sensor (which is integrated in the unit, so will always give a higher temp than the actual temp in the room).

**Comment by BeeGee1986** [comment]
> That's why I created my own automation. If I want it to be 20 degrees C, I'll set it to heat and to 23 or 24 degrees. Then I have another automation that will stop the airco if the external temp sensor reaches 20 (actually, I use the average temp, from 2 temp sensors in my room). The first automation will start it when the temp gets below 18, so this way the unit will only be on until the temp is reached and I bypass Daikin logic that way, and also their temp sensor (which is integrated in the unit, so will always give a higher temp than the actual temp in the room).

That is really interesting. Would you be able to share the automation or parts of it? I'm curious to see how you have structured it and might "steal" some logic of it to give this a try :-)

**Comment by rgerhards** [comment]
I have a similar appraoch as @OcinO88, but I admit I have not yet implemented it (works currently sufficiently well for my use case with Daikin only, so I left the implementation for when I have time... well...).

The basic idea is exactly as described. I just intend to change set target temp up/down. I would expect that this is somewhat more friendly to the units (think: compressor operation) than to tell it to shut down and start up. Not sure if it really makes a difference, And, if so, I'd say its a small diff. Just wanted to share this thought.

**Comment by OcinO88** [comment]
It is something like this (for heating my living room, I added that my phone needs to be home, so it will only heat when I'm here and it's needed. I also created an automation, when arriving at home, to check for the temp and set the airco to cool or heat, depending on what's needed.):
![image](https://github.com/rospogrigio/daikin_residential/assets/90789065/b657f4ce-fc06-4d80-909e-50dac66f3b6b)
![image](https://github.com/rospogrigio/daikin_residential/assets/90789065/d2a0f178-f2aa-4c2b-adf9-67d5c8d8531a)
![image](https://github.com/rospogrigio/daikin_residential/assets/90789065/c74b8280-fec1-46e5-97a6-35219c874946)
![image](https://github.com/rospogrigio/daikin_residential/assets/90789065/f6500d59-6492-49ff-930a-8fa648ac70f7)

To stop heating:
![image](https://github.com/rospogrigio/daikin_residential/assets/90789065/a7d4f5b1-44d6-465b-bb38-479161b9e97d)
![image](https://github.com/rospogrigio/daikin_residential/assets/90789065/7f1c74db-5a9f-4846-82d7-d47f9786ea9b)
